### PR TITLE
[ComputerCraft] Add new lua functions to Bank Terminal Peripheral

### PR DESCRIPTION
**Pull request by banna12345bob**
Adds four new lua functions to the Bank Terminal peripheral.
- `getAccounts()`
- `getAccountLabel(String accountID)`
- `getSubAccounts(String accountID)`
- `getSubAccountLabel(String accountID, String authorizationID)`

## `getAccounts()`
Gets a list of UUIDs for all accounts.
#### **Returns**
 - `list` of UUIDs for all accounts.
## `getAccountLabel(String accountID)`
Get the label of the specified account.
Throws an error if the account is not found.
#### **Returns**
- `string` the name of the specified account.
## `getSubAccounts(String accountID)`
Gives a list of UUIDs for any sub-accounts inside of specified account.
Only displays sub-accounts with authorisation type set to `Anybody`.
#### **Returns**
- `list` of UUIDs for sub-accounts
### `getSubAccountLabel(String accountID, String authorizationID)`
Returns the name for the specified sub-account inside of the specified account.
Authorisation type must be set to `Anybody` otherwise will throw an error.
#### **Returns**
- `string` the name of the specified sub-account.

### CODE REVIEW

**Comment by techno-sam** [comment]
I don't think it's a good idea to return a list of *any* sub accounts, because someone might collect IDs and wait till they're changed to Anybody

**Comment by banna12345bob** [comment]
What function are you referring to? The `getSubAccounts` function only returns accounts set to `Anybody`.

**Comment by techno-sam** [comment]
That's exactly my point. I don't want people to be able to get a list of `Anybody` accounts, because those should still have the security afforded by "I was careful not to give anyone I don't trust the UUID"

**Comment by banna12345bob** [comment]
How do you intend users to get uuids of accounts that the owner wants them to access? Would it be preferable to put the uuids in the accounts/sub-accounts tab? 
How would you expect users to remember all of their account uuids?

There is always a chance that malicious users read the uuid out of a already programmed computer thus rendering any security steps useless.

If you wish to contact me further about any issues with this PR I am `.constantly` on discord.

**Comment by techno-sam** [comment]
Users are intended to get the UUIDs by using a computer to read them from an authorized card. Of course, malicious users could read the uuid from an *unsecured* computer, but, as long as the relevant server has some sort of claims mod, computers could be kept secure.

Other than enumerating sub accounts, I'm for merging this PR

**Review by IThundxr — APPROVED** [review]

File: common/src/main/java/dev/ithundxr/createnumismatics/compat/computercraft/implementation/peripherals/BankTerminalPeripheral.java [hunk 1]
```diff
@@ -32,12 +32,38 @@
...
+            throw new LuaException("Invalid UUID");
+        }
+
+        BankAccount bankAccount = Numismatics.BANK.getAccount(account$);
+        if (bankAccount == null) {
+            throw new LuaException("Account not found");
+        }
+
+        return bankAccount.getDisplayName().getString();
+    }
```
Comment (techno-sam) [inline common/src/main/java/dev/ithundxr/createnumismatics/compat/computercraft/implementation/peripherals/BankTerminalPeripheral.java:66]: would it be helpful to also add a function that returns whether an account is player- or blaze banker-owned?

File: common/src/main/java/dev/ithundxr/createnumismatics/compat/computercraft/implementation/peripherals/BankTerminalPeripheral.java [hunk 2]
```diff
@@ -55,6 +81,61 @@ public final int getBalance(String accountID) throws LuaException {
         return bankAccount.getBalance();
     }
 
+    @LuaFunction
+    public final List<String> getSubAccounts(String accountID) throws LuaException {
```
Comment (techno-sam) [inline common/src/main/java/dev/ithundxr/createnumismatics/compat/computercraft/implementation/peripherals/BankTerminalPeripheral.java]: This function should be removed or disabled by default (subject to config), see thread above for reasons why

**Review by techno-sam — CHANGES_REQUESTED** [review]
Couple of nitpicks, but very good!

File: common/src/main/java/dev/ithundxr/createnumismatics/config/CServer.java [hunk 3]
```diff
@@ -33,6 +33,10 @@ public class CServer extends ConfigBase {
     public final ConfigInt starterCogs = i(0, 0, "starter_cogs");
     public final ConfigInt starterCrowns = i(0, 0, "starter_crowns");
     public final ConfigInt starterSuns = i(0, 0, "starter_suns");
+
+    public final ConfigGroup computerCraft = group(0, "computerCraft", Comments.computerCraft);
+
+    public final ConfigBool getSubAccountsCommand = b(false, "get_sub_accounts_command");
```
Comment (techno-sam) [inline common/src/main/java/dev/ithundxr/createnumismatics/config/CServer.java]: This needs a description, then good to merge

**Review by techno-sam — CHANGES_REQUESTED** [review]

**Comment by banna12345bob** [comment]
Documentation for new function.
## `isPlayerOwned(String accountID)`
Whether a player owns the specified account or not.
#### **Returns**
 - `bool` true if account is player-owned, false if not.

**Review by techno-sam — APPROVED** [review]
lgtm

**Pull request merged**